2022/Q4/artifacts/PoS-pseudocode/PoS-model.md

```diff
@@ -240,10 +248,12 @@ func bond(validator_address, delegator_address, amount)
     //debit amount from delegator account and credit it to the PoS account
     balances[delegator_address] -= amount
     balances[pos] += amount
-    update_total_deltas(validator_address, pipeline_lenght, amount)
-    update_voting_power(validator_address, pipeline_lenght)
-    update_total_voting_power(pipeline_lenght)
-    update_validator_sets(validator_address, pipeline_lenght)
+    update_total_deltas(validator_address, pipeline_length, amount)
+    update_voting_power(validator_address, pipeline_length)
+    update_total_voting_power(pipeline_length)
+    // Only update validator sets if the validator is not jailed, since a jailed validator is not in a validator set
+    if !is_jailed(validator_address) then
```

Comment:
This is not needed, we remove a validator from active and inactive sets when jailing it.